### PR TITLE
feat: Update facade for Unit children and cost centers + views

### LINKED ISSUE #6561: Prepare Occ & Core functionality for Cost centers for Unit

### PR DESCRIPTION
**Pull request by Pucek9**
Closes #6561 #6562 #6991 #6992

### CODE REVIEW

File: projects/core/src/organization/facade/org-unit.service.ts [hunk 1]
```diff
@@ -133,6 +134,28 @@ export class OrgUnitService {
...
+
+  protected findUnitChildrenInTree(orginitId, unit: B2BUnitNode) {
+    if (unit.id === orginitId) {
+      return unit.children;
+    } else {
+      return unit.children.flatMap((child) =>
+        this.findUnitChildrenInTree(orginitId, child)
+      );
+    }
+  }
```
Comment (Zeyber) [inline projects/core/src/organization/facade/org-unit.service.ts]:
```suggestion
  protected findUnitChildrenInTree(orginitId, unit: B2BUnitNode) {
    return unit.id === orginitId
      ? unit.children
      : unit.children.flatMap((child) =>
          this.findUnitChildrenInTree(orginitId, child)
        );
  }
```

File: projects/core/src/organization/facade/org-unit.service.ts [hunk 2]
```diff
@@ -133,6 +134,28 @@ export class OrgUnitService {
...
+
+  getChildUnits(orgUnitId: string): Observable<B2BUnitNode[]> {
+    return this.getTree().pipe(
+      map((tree) => {
+        return this.findUnitChildrenInTree(orgUnitId, tree);
+      })
+    );
```
Comment (Zeyber) [inline projects/core/src/organization/facade/org-unit.service.ts]:
```suggestion
    return this.getTree().pipe(
      map((tree) => this.findUnitChildrenInTree(orgUnitId, tree))
    );
```

**Review by Zeyber — CHANGES_REQUESTED** [review]
Child Unit issues found:
- [x] Create a child unit named Test A and then try to create another child unit from Test A called Test B. Test A will not appear in the `Parent business unit` dropdown on the `Create Unit` view until page refresh.
- [x] Rename parent unit and navigate to edit child unit. Parent unit is unlabeled in dropdown list.
- [ ] Cancel `Edit Unit` view navigates to `Units` view instead of `Unit Details` view.
- [ ] No option to delete units (could be by design)

Cost Center Issues found:
- [x] Cannot enable cost center `Services_East`. Click enable and the request is sent and the server responds with `active: true` but the status still shows disabled in Cost Center details even after refresh. Same issue with all cost centers. Can't test the rest of cost centers until this is resolved.

Note: Tested against dev-7

**Comment by Pucek9** [comment]
> Child Unit issues found:
> 
> * [x]  Create a child unit named Test A and then try to create another child unit from Test A called Test B. Test A will not appear in the `Parent business unit` dropdown on the `Create Unit` view until page refresh.
> * [x]  Rename parent unit and navigate to edit child unit. Parent unit is unlabeled in dropdown list.

Fixed by force load buisness units in form
> * [ ]  Cancel `Edit Unit` view navigates to `Units` view instead of `Unit Details` view.

Will fix it globally when we finally establish back button functionality
> * [ ]  No option to delete units (could be by design)

It's not in the requirements
> 
> Cost Center Issues found:
> 
> * [x]  Cannot enable cost center `Services_East`. Click enable and the request is sent and the server responds with `active: true` but the status still shows disabled in Cost Center details even after refresh. Same issue with all cost centers. Can't test the rest of cost centers until this is resolved.

As it was tested last time, update costcenters was correct (enabled parent)
https://cxwiki.sap.com/display/spar/Behavior+of+entities%2C+which+parent+is+disabled
Need to create C3PO issue that we are not getting activeFlag anymore and got `active: "true"` instead of boolean
**UPDATE**: temporary fixed in views by use existing string value `costCenter.active === 'true'` after discuss with Frank

> 
> Note: Tested against dev-7

**Review by mszczepaniak — APPROVED** [review]

**Pull request merged**